### PR TITLE
[NPM-2843] adds benchmark test for process_cache

### PR DESCRIPTION
**Pull request by CordeliaLopez**
### What does this PR do?
This PR adds a Benchmark test to see the effects of code changes in allocations for the cache

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by hmahmood — APPROVED** [review]

**Review by heyronhay — APPROVED** [review]

File: pkg/network/tracer/process_cache_test.go [hunk 1]
```diff
@@ -339,3 +340,24 @@ func TestProcessCacheGet(t *testing.T) {
 	}
 
 }
+
+func BenchmarkProcessCacheMem(b *testing.B) {
+	pc, _ := newProcessCache(b.N, nil)
```
Comment (leeavital) [inline pkg/network/tracer/process_cache_test.go:345]: I think you want to use a constant size here -- why did you use b.N?

Comment (CordeliaLopez) [reply pkg/network/tracer/process_cache_test.go:345]: @hmahmood suggested it to see the cache increase when b.N increased. We tried a few numbers. Do you have a suggestion as to what the size might be?

Comment (leeavital) [reply pkg/network/tracer/process_cache_test.go:345]: > it to see the cache increase when b.N increased

I think I might be missing something, but is that what we want?

I would suggest any constant, it doesn't really matter what. 

In a benchmark, the thing inside the `for i := 0; i < b.N; i++` should have a constant cost no matter what `b.N` is.

Comment (hmahmood) [reply pkg/network/tracer/process_cache_test.go:345]: Its an LRU, so if `b.N > size`, we'll see evicts. I just wanted to eliminate that possibility.

**Pull request closed without merging**